### PR TITLE
Form config bug fixes

### PR DESCRIPTION
**Pull request by euanmillar**

### CODE REVIEW

File: packages/client/src/components/formConfig/SectionNavigation.tsx [hunk 1]
```diff
@@ -54,9 +54,14 @@ const PageItems = styled(NavigationSubItem)<{ isSelected: boolean }>`
 function previewSectionFilter(
   section: BirthSection | DeathSection
 ): section is
-  | Exclude<BirthSection, 'preview'>
-  | Exclude<DeathSection, 'preview'> {
-  return section !== BirthSection.Preview && section !== DeathSection.Preview
+  | Exclude<BirthSection, 'preview' | 'registration'>
+  | Exclude<DeathSection, 'preview' | 'registration'> {
```
Comment (euanmillar) [inline packages/client/src/components/formConfig/SectionNavigation.tsx:58]: I think we should hide registration section entirely from configuration

File: packages/client/src/forms/configuration/defaultUtils.ts [hunk 2]
```diff
@@ -71,21 +71,30 @@ export function getDefaultField(
 
 export function configureDefaultQuestions(
   defaultFieldCustomisations: IDefaultFieldCustomisation[],
-  defaultEventForm: ISerializedForm
+  defaultEventForm: ISerializedForm,
+  inConfig: boolean
```
Comment (euanmillar) [inline packages/client/src/forms/configuration/defaultUtils.ts:75]: the inConfig boolean allows us to splice disabled fields away from the form only when in registration, not configuration mode.

File: packages/client/src/i18n/messages/views/formConfig.ts [hunk 3]
```diff
@@ -129,15 +131,16 @@ const navigationMessagesToDefine: INavigationMessages = {
...
-    description: 'Label for registration in page navigation'
   }
 }
 
 const messagesToDefine: IFormConfigMessages = {
+  globalSettingsDescription: {
+    id: 'config.form.settings.globalSettingsDescription',
+    defaultMessage:
+      'These are global form settings that can be configured.  Please note that applying any changes to these settings takes effect immediately to the currently active forms independently to your draft status.',
+    description: 'Label for global form settings introduction'
```
Comment (euanmillar) [inline packages/client/src/i18n/messages/views/formConfig.ts:142]: We need to inform the user that changing these settings has an immediate impact as we are updating application config not form config

File: packages/client/src/forms/configuration/defaultUtils.ts [hunk 4]
```diff
@@ -71,21 +71,30 @@ export function getDefaultField(
...
-    /*
-     * Splicing the field away may not be the way to go as it's
-     * removing the field from FormConfig as well
-     */
+    const field: SerializedFormField =
+      newForm.sections[
+        defaultFieldCustomisation.defaultField.selectedSectionIndex
+      ].groups[defaultFieldCustomisation.defaultField.selectedGroupIndex]
+        .fields[defaultFieldCustomisation.defaultField.index]
+    field.required = defaultFieldCustomisation.question.required
```
Comment (euanmillar) [inline packages/client/src/forms/configuration/defaultUtils.ts:88]: configures the required value

File: packages/client/src/views/SysAdmin/Config/Forms/Wizard/FormConfigWizard.tsx [hunk 5]
```diff
@@ -261,13 +276,16 @@ export function FormConfigWizard() {
                     event={event}
                     section={section}
                     selectedField={selectedField}
+                    setSelectedField={setSelectedField}
+                    groupId={firstFieldIdentifiers.groupId}
```
Comment (Zangetsu101) [inline packages/client/src/views/SysAdmin/Config/Forms/Wizard/FormConfigWizard.tsx:280]: Should the group be the first one in the section or the last one?

**Review by Zangetsu101 — COMMENTED** [review]
What was the reason for adding identifiers to `CustomConfigFields`? The identifiers were only needed for `DefaultConfigFields` to get their field definition from the `configRegisterForm`(we cannot use the default `registerForm`), for `CustomConfigFields` we created their field definition. So I don't see where they are needed.

**Comment by Zangetsu101** [comment]
Closing this in favor of #3015

**Pull request closed without merging**